### PR TITLE
Remove `setController` Method

### PR DESCRIPTION
**Pull request by DonJayamanne**
Something that came out of another PR,

### CODE REVIEW

File: src/client/datascience/notebook/notebookControllerManager.ts [hunk 1]
```diff
@@ -187,53 +187,43 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
...
-        } else {
-            this.setController(document, preferredSearchToken.token).finally(() => {
-                // Make sure that we clear our finding in progress when done
-                this.findPreferredInProgress.delete(document);
-            });
-        }
-    }
-
-    // Set the controller for this notebook document
-    private async setController(document: NotebookDocument, cancelToken: CancellationToken) {
```
Comment (DonJayamanne) [inline src/client/datascience/notebook/notebookControllerManager.ts:211]: This method doesn't set the controller at all, if anything it finds the preferred controller and sets the preferred controller.
We also have an existing method named `setPreferredController`.
Also, it wasn't clear to me my we were clearing ` this.controllerMapping.delete(document);` in line `214`.

I felt that it makes more sense to have this code inline in the event handler. 
To me the code reads better, but if others disagree please reject this PR

**Again, this is NOT a required PR, just a simple codehealth/refactor.**

Comment (rchiodo) [reply src/client/datascience/notebook/notebookControllerManager.ts:211]: Looks good to me. Should probably wait for Ian as he wrote this originally.

Comment (IanMatthewHuff) [reply src/client/datascience/notebook/notebookControllerManager.ts:211]: The main reason that this function was structured as it is, is that I believe we discussed wanted to keep the onDidOpen / onDidClose sync and not have awaits in them. However I can't track down that initial discussion on that. If aync on the doc functions is not a problem I'd prefer this as well.

Comment (DonJayamanne) [reply src/client/datascience/notebook/notebookControllerManager.ts:211]: Looks like we're ok, please can we get this approved.

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**